src/config/tenants/tenant.rs

```diff
@@ -160,6 +165,27 @@ pub async fn delete_tenant_ui(db: &Database) -> Result<(), BiError> {
         Ok(num) if num > 0 && num <= tenants_with_realms.len() => {
             // Find selected tenant and realm
             let (tenant, realm) = &tenants_with_realms[num - 1];
+
+            // Check if this is the default tenant
+            let is_default = match db.get_default_tenant_and_realm().await? {
+                Some((default_tenant, default_realm)) => {
+                    default_tenant.id == tenant.id && default_realm.id == realm.id
+                }
+                None => false,
+            };
+
+            if is_default {
+                println!("Warning: You are about to delete the default tenant/realm.");
+                print!("Do you want to continue? (y/n): ");
+                io::stdout().flush().unwrap();
+                let mut confirm = String::new();
+                io::stdin().read_line(&mut confirm).unwrap();
```

Comment:
Not the biggest deal in this context but with `unwrap` due to the underlying implementation `io::stdin` uses which can be read about here: https://doc.rust-lang.org/std/io/trait.BufRead.html#method.read_line

Since we dont have error handling within the tool at this time I think its fine, and none of the panic scenerios should cause data corruption so not a huge deal. We are just pretty exposed to panicing here due to user error / attempting to crash the processes through buffer overdflows.